app/components/homepage-components/ServicesCard.tsx

```diff
@@ -1,18 +1,21 @@
 "use client";
 
-import { useState } from "react";
-import Image from "next/image";
+import { motion } from "framer-motion";
 import { useTheme } from "next-themes";
+import { useState, useEffect } from "react";
+import Image from "next/image";
 
 import { ServiceProvidedType } from "@/types";
 interface ServicesCardProps {
   service: ServiceProvidedType;
+  delay?: number;
 }
 
-const ServicesCard = ({ service }: ServicesCardProps) => {
+const ServicesCard = ({ service, delay }: ServicesCardProps) => {
   const { theme } = useTheme();
   const [isMouseOver, setIsMouseOver] = useState<boolean>(false);
-
+  const [backgroundColor, setBackgroundColor] = useState("");
+  const [hoverBackgroundColour, setHoverBackgroundColour] = useState("");
   let imageSrc;
 
   if (isMouseOver) {
```

Comment:
What you have works perfectly... No problems and great job! I want to give you another approach to consider. The logic might be harder to read but look it over and see if it makes sense and you want to implement the approach. Just an option...

```
const imageSrc = isMouseOver
  ? theme === "light"
    ? service.imageHoverLight
    : service.imageHoverDark
  : service.imageSrc;
```
And inside your Image, you could refer to it as:
```
<Image
  src={imageSrc.image}
  height={24}
  width={24}
  alt={`${service.title} icon`} // I snuck in "icon" as it provides a little more detail. :)
/>
```

Reply:
This is way better. Thanks for the suggestion!